From 442ab414f7314e813aafbbaa29e3363d6777e885 Mon Sep 17 00:00:00 2001
From: Yuya Hamamachi <yuya.hamamachi.sx@renesas.com>
Date: Wed, 28 Oct 2026 20:35:07 +0900
Subject: [PATCH 12/15] archiver: Archive GPL sources for SBOM release

create-spdx does not output the source of gcc, because it skips the gcc
recipes and gcc-source has no file information. Then the source of the
GPL/LGPL packages could not be provided with the release.

Enable the archiver class together with the SBOM build so that the
original source, the patches and the recipe of each copyleft package
are archived into tmp/deploy/sources, and add them to the release
assets as deploy_sources_split.zip.

Drop SPDX_ARCHIVE_SOURCES to avoid providing the source archives by two
mechanisms. The file information of the sources is still included in
the SPDX by SPDX_INCLUDE_SOURCES.

Signed-off-by: Yuya Hamamachi <yuya.hamamachi.sx@renesas.com>
---
 .github/workflows/release.yaml |  2 ++
 build.sh                       | 12 +++++++++---
 2 files changed, 11 insertions(+), 3 deletions(-)

diff --git a/.github/workflows/release.yaml b/.github/workflows/release.yaml
index 0084941..319d34a 100644
--- a/.github/workflows/release.yaml
+++ b/.github/workflows/release.yaml
@@ -43,6 +43,7 @@ jobs:
           zip -r ${GITHUB_WORKSPACE}/build/build-sparrow-hawk/tmp/deploy/ipl-burning.zip ./ipl-burning
           cd ${GITHUB_WORKSPACE}/build/build-sparrow-hawk/tmp/deploy
           zip -ry -s 1900m deploy_spdx_split.zip spdx/
+          zip -ry -s 1900m deploy_sources_split.zip sources/
           cd $GITHUB_WORKSPACE
 
       - name: Prepare release message
@@ -173,6 +174,7 @@ jobs:
           files: |
             ./build/build-sparrow-hawk/tmp/deploy/images/deploy_sparrow-hawk.z*
             ./build/build-sparrow-hawk/tmp/deploy/deploy_spdx_split.z*
+            ./build/build-sparrow-hawk/tmp/deploy/deploy_sources_split.z*
             ./build/build-sparrow-hawk/tmp/deploy/ipl-burning.zip
             ./build/build-sparrow-hawk/tmp/deploy/images/*/core-image-*-sparrow-hawk.rootfs.wic.*
           body_path: ${{ github.workspace }}-release_msg.txt
diff --git a/build.sh b/build.sh
index 02869c1..3e8c9ef 100755
--- a/build.sh
+++ b/build.sh
@@ -114,11 +114,17 @@ SPDX_PRETTY = "1"
 # optional. if "1", output spdx files includes [file-information section](https://spdx.github.io/spdx-spec/v2.3/file-information/).
 SPDX_INCLUDE_SOURCES = "1"
 
-# optional. if "1", bitbake will create source files archive for each package.
-SPDX_ARCHIVE_SOURCES = "1"
-
 # optional. if "1", bitbake will create output binary archive for each package.
 SPDX_ARCHIVE_PACKAGED = "1"
+
+# required. enable to archive the source of GPL/LGPL/AGPL packages
+# into tmp/deploy/sources.
+INHERIT += "archiver"
+
+# optional. archive the original source, patches and recipe.
+ARCHIVER_MODE[src] = "original"
+ARCHIVER_MODE[diff] = "1"
+ARCHIVER_MODE[recipe] = "1"
 EOS
 else # Disable SBOM build to reduce build time
 cat << EOS >> conf/local.conf
